Tests/SwiftRefactorTest/ManifestEditTests.swift

```diff
@@ -0,0 +1,728 @@
+              .target(
+                  name: "MyLib"
+              )
+          ]
+      )
+      """,
+      expectedManifest: """
+        // swift-tools-version: 5.7
+        let package = Package(
+            name: "packages",
+            targets: [
+                .target(
+                    name: "MyLib",
+                    plugins: [
+                        .plugin(name: "Java2SwiftPlugin", package: "swift-java"),
+                    ]
+                )
+            ]
+        )
+        """,
+      provider: AddPluginUsage.self,
+      context: .init(
+        targetName: "MyLib",
+        pluginUsage: .plugin(name: "Java2SwiftPlugin", package: "swift-java")
+      )
+    )
+  }
+}
+
+/// Assert that applying the given edit/refactor operation to the manifest
+/// produces the expected manifest source file and the expected auxiliary
+/// files.
+func assertManifestRefactor<Provider: ManifestEditRefactoringProvider>(
+  _ originalManifest: SourceFileSyntax,
+  expectedManifest: SourceFileSyntax,
+  expectedAuxiliarySources: [RelativePath: SourceFileSyntax] = [:],
+  provider: Provider.Type,
+  context: Provider.Context,
+  file: StaticString = #filePath,
+  line: UInt = #line
+) throws {
+  return try assertManifestRefactor(
+    originalManifest,
+    expectedManifest: expectedManifest,
+    expectedAuxiliarySources: expectedAuxiliarySources,
+    file: file,
+    line: line
+  ) { (manifest) in
+    try provider.manifestRefactor(syntax: manifest, in: context)
+  }
+}
+
+/// Assert that applying the given edit/refactor operation to the manifest
+/// produces the expected manifest source file and the expected auxiliary
+/// files.
+func assertManifestRefactor(
+  _ originalManifest: SourceFileSyntax,
+  expectedManifest: SourceFileSyntax,
+  expectedAuxiliarySources: [RelativePath: SourceFileSyntax] = [:],
+  file: StaticString = #filePath,
+  line: UInt = #line,
+  operation: (SourceFileSyntax) throws -> PackageEdit
+) rethrows {
+  let edits = try operation(originalManifest)
+  let editedManifestSource = FixItApplier.apply(
+    edits: edits.manifestEdits,
+    to: originalManifest
+  )
+
+  let editedManifest = Parser.parse(source: editedManifestSource)
+  assertStringsEqualWithDiff(
+    editedManifest.description,
+    expectedManifest.description,
+    file: file,
+    line: line
+  )
+
+  // Check all of the auxiliary sources.
+  for (auxSourcePath, auxSourceSyntax) in edits.auxiliaryFiles {
+    guard let expectedSyntax = expectedAuxiliarySources[auxSourcePath] else {
+      XCTFail("unexpected auxiliary source file \(auxSourcePath)")
+      return
+    }
+
+    assertStringsEqualWithDiff(
+      auxSourceSyntax.description,
+      expectedSyntax.description,
+      file: file,
+      line: line
+    )
+  }
+
+  XCTAssertEqual(
+    edits.auxiliaryFiles.count,
+    expectedAuxiliarySources.count,
+    "didn't get all of the auxiliary files we expected"
+  )
+}
+
+func XCTAssertThrows<T: Swift.Error, Ignore>(
```

Comment:
Why doesn’t `XCTAssertThrowsError` from XCTest work here? If it doesn’t work, I would name this `assertThrows` because we don’t want to define local functions with the `XCT` prefix. Also, I would make this `fileprivate` to avoid it leaking into other tests.